crates/core/c8y_api/src/http_proxy.rs

```diff
@@ -89,29 +89,19 @@ impl C8yEndPoint {
         url_event_binary
     }
 
-    pub fn url_is_in_my_tenant_domain(&self, url: &str) -> bool {
+    pub fn maybe_tenant_url(&self, url: &str) -> Option<Url> {
```

Comment:
> So, that name with a bool return felt more natural.

The issue with functions, that check a shape and return a bool, is that if you want to go further you have to redo the parsing work a second time dealing with error cases which should no more happens.

If you only need a check this can be done by chaining `self.maybe_tenant_url(url).is_some()`.